packages/theming-core/src/generics/themeRegistry.test.ts

```diff
@@ -0,0 +1,125 @@
+import { createThemeRegistry, IThemeRegistry } from './themeRegistry';
+
+interface IPartialTestTheme {
+  val: number;
+}
+
+interface ITestTheme {
+  val: number;
+  sum: number;
+}
+
+function _resolveTestTheme(definition: IPartialTestTheme | undefined, parent: ITestTheme): ITestTheme {
```

Comment:
Similar to Jason's comment below, you could consider the following to get rid of the requirement of passing `definition`:

```suggestion
function _resolveTestTheme(parent: ITestTheme, definition?: IPartialTestTheme): ITestTheme {
```